### PR TITLE
Fix mypy for `utilities.xla_device`

### PR DESCRIPTION
**Pull request by stancld**
## What does this PR do?

Part of #7037. This PR disentangles large #7218. It fixes mypy typing in `utilities.xla_device.py`.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/utilities/imports.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def _compare_version(package: str, op, version) -> bool:
 _TORCHVISION_AVAILABLE = _module_available("torchvision")
 _TORCHMETRICS_LOWER_THAN_0_3 = _compare_version("torchmetrics", operator.lt, "0.3.0")
 _TORCHMETRICS_GREATER_EQUAL_0_3 = _compare_version("torchmetrics", operator.ge, "0.3.0")
-_XLA_AVAILABLE = _module_available("torch_xla")
+_XLA_AVAILABLE: bool = _module_available("torch_xla")
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:92]: Why is this necessary?

`_module_available` is annotated with `-> bool`

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/utilities/imports.py (hunk 1 above)
Comment (stancld) [reply pytorch_lightning/utilities/imports.py:92]: Unfortunately, mypy is not able to figure that out without this explicit type hint.

Comment (SkafteNicki) [reply pytorch_lightning/utilities/imports.py:92]: can confirm, we had the same problem in metrics:
https://github.com/PyTorchLightning/metrics/blob/525642df8c3f3f5f93cdd22b6b9cd802a3854634/torchmetrics/utilities/imports.py#L79-L80

**Review by SkafteNicki — APPROVED** [review]
